### PR TITLE
feat: add fixable, no more error when unused variable with prefix _

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
implement proposal https://github.com/vuejs/eslint-plugin-vue/issues/1058

feat https://github.com/vuejs/eslint-plugin-vue/issues/1058

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
okay, i will do that.

**Comment by IWANABETHATGUY** [comment]
> Thank you for this PR.
> 
> I like to allow users to ignore variables with specific names.
> 
> But, I think it should be an option like `varsIgnorePattern`, just like ESLint's [`no-unused-vars`](https://eslint.org/docs/rules/no-unused-vars).
> 
> https://eslint.org/docs/rules/no-unused-vars
> 
> The `vue/no-unused-vars` rule cannot distinguish between `varsIgnorePattern` and `argsIgnorePattern`, so I think it's better to add `ignorePattern` option.
> 
> Also, if we change the variable name by auto fix, it may conflict with other variables. So, could you change it so that it does not autofix it.

Do you mean only add the `varIgnorePattern` , don't provide fixable option?

File: lib/rules/no-unused-vars.js [hunk 1]
```diff
@@ -18,26 +18,49 @@ module.exports = {
...
+      }
+    ]
   },
 
   create (context) {
+    const option = context.options[0] || { varIgnorePattern: '^_' }
+    const pattern = option['varIgnorePattern'] || '^_'
+    // only use for construct a regularExpression
+    // eslint-disable-next-line no-eval
+    const regExp = eval(`/${pattern}/`)
```
Comment (ota-meshi) [inline lib/rules/no-unused-vars.js]: Do not use `eval()`.

You can handle it in the same way as ESLint's no-unused-vars.

https://github.com/eslint/eslint/blob/master/lib/rules/no-unused-vars.js#L106

File: lib/rules/no-unused-vars.js [hunk 2]
```diff
@@ -18,26 +18,49 @@ module.exports = {
...
       url: 'https://eslint.vuejs.org/rules/no-unused-vars.html'
     },
-    fixable: null,
-    schema: []
+    fixable: 'code',
+    schema: [
+      {
+        'type': 'object',
+        'properties': {
+          'varIgnorePattern': {
```
Comment (ota-meshi) [inline lib/rules/no-unused-vars.js]: Could you change the name to `ignorePattern`?

File: lib/rules/no-unused-vars.js [hunk 3]
```diff
@@ -18,26 +18,49 @@ module.exports = {
...
+            'type': 'string'
+          }
+        },
+        'additionalProperties': false
+      }
+    ]
   },
 
   create (context) {
+    const option = context.options[0] || { varIgnorePattern: '^_' }
```
Comment (ota-meshi) [inline lib/rules/no-unused-vars.js]: If user don't set option, could you change the default behavior to not ignore everything?

File: lib/rules/no-unused-vars.js [hunk 4]
```diff
@@ -18,26 +18,49 @@ module.exports = {
...
             node: variable.id,
             loc: variable.id.loc,
             message: `'{{name}}' is defined but never used.`,
-            data: variable.id
+            data: variable.id,
+            suggest: [
+              {
+                desc: 'Replace the unused-var with default ignore pattern',
+                fix: function (fixer) {
+                  return fixer.replaceText(variable.id, `_${variable.id.name}`)
```
Comment (ota-meshi) [inline lib/rules/no-unused-vars.js]: I think it is good to use SuggestionAPI.

However, this suggestion is only valid if `'^_'` is set.

File: tests/lib/rules/no-unused-vars.js [hunk 5]
```diff
@@ -98,6 +115,21 @@ tester.run('no-unused-vars', rule, {
...
+    },
+    {
+      code: '<template><span slot-scope="props"></span></template>',
+      errors: ["'props' is defined but never used."],
+      options: [{ varIgnorePattern: '^ignore' }]
+    },
+    {
+      code: '<template><span><template scope="props"></template></span></template>',
+      errors: ["'props' is defined but never used."],
+      options: [{ varIgnorePattern: '^ignore' }]
```
Comment (ota-meshi) [inline tests/lib/rules/no-unused-vars.js]: Could you add test cases for suggestions?

https://eslint.org/docs/developer-guide/nodejs-api#testing-suggestions

File: docs/rules/no-unused-vars.md [hunk 6]
```diff
@@ -8,12 +8,13 @@ description: disallow unused variable definitions of v-for directives or scope a
 > disallow unused variable definitions of v-for directives or scope attributes
 
 - :gear: This rule is included in all of `"plugin:vue/essential"`, `"plugin:vue/strongly-recommended"` and `"plugin:vue/recommended"`.
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
 
 ## :book: Rule Details
 
 This rule report variable definitions of v-for directives or scope attributes if those are not used.
```
Comment (ota-meshi) [inline docs/rules/no-unused-vars.md]: Could you add options document.

**Comment by IWANABETHATGUY** [comment]
> LGTM! Thank you very match!
😀

**Pull request merged**